### PR TITLE
Ra 1122

### PR DESCRIPTION
**Pull request by VoldarGames**

### CODE REVIEW

File: VAS.Core/MVVMC/PropertyBinding.cs [hunk 1]
```diff
@@ -139,24 +130,71 @@ protected override void UnbindViewModel ()
...
+	}
+
+	/// <summary>
+	/// This class provides a default type converter for equal types.
+	/// </summary>
+	class DefaultTypeConverter<TSourceProperty, TTargetProperty> : TypeConverter
+	{
+		public override bool CanConvertFrom (ITypeDescriptorContext context, Type sourceType)
+		{
+			return (typeof (TTargetProperty).IsAssignableFrom (typeof (TSourceProperty)));
```
Comment (ylatuya) [inline VAS.Core/MVVMC/PropertyBinding.cs]: For the converter to work on both directions TTargetProperty and TSourceProperty must be the same type

File: VAS.Tests/MVVMC/TestOneWayPropertyBinding.cs [hunk 2]
```diff
@@ -24,6 +25,30 @@
 
 namespace VAS.Tests.MVVMC
 {
+
+	class IntegerParserConverter : TypeConverter
```
Comment (ylatuya) [inline VAS.Tests/MVVMC/TestOneWayPropertyBinding.cs]: Can't you use the Int32Converter directly?

File: VAS.UI.Helpers.Gtk2/Bindings/SpinBinding.cs [hunk 3]
```diff
@@ -27,12 +27,12 @@ namespace VAS.UI.Helpers.Bindings
 	/// <summary>
 	/// Property binding for the spin value changes
 	/// </summary>
-	public class SpinBinding : PropertyBinding<double>
+	public class SpinBinding<T> : PropertyBinding<T, double>
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/SpinBinding.cs]: T should have meaningful name

File: VAS.UI.Helpers.Gtk2/Bindings/LabelBinding.cs [hunk 4]
```diff
@@ -27,16 +27,11 @@ namespace VAS.UI.Helpers.Bindings
 	/// <summary>
 	/// Property bindings for labels.
 	/// </summary>
-	public class LabelBinding : PropertyBinding<string>
+	public class LabelBinding<T> : PropertyBinding<T, string>
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/LabelBinding.cs]: T should have meaningful name

File: VAS.UI.Helpers.Gtk2/Bindings/ComboBoxBinding.cs [hunk 5]
```diff
@@ -27,16 +27,12 @@ namespace VAS.UI.Helpers.Bindings
 	/// <summary>
 	/// Property binding for text entries.
 	/// </summary>
-	public class ComboBoxBinding : PropertyBinding<string>
+	public class ComboBoxBinding<T> : PropertyBinding<T, string>
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/ComboBoxBinding.cs]: T should have meaningful name

File: VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs [hunk 6]
```diff
@@ -47,9 +47,9 @@ public static BindingContext GetBindingContext (this Widget widget)
 		/// Bind the specified label to a property by name.
 		/// </summary>
 		/// <param name="label">Label.</param>
-		public static LabelBinding Bind (this Label label, Expression<Func<IViewModel, string>> propertyExpression)
+		public static LabelBinding<T> Bind<T> (this Label label, Expression<Func<IViewModel, T>> propertyExpression)
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs]: T should have meaningful name

File: VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs [hunk 7]
```diff
@@ -59,9 +59,9 @@ public static LabelBinding Bind (this Label label, Expression<Func<IViewModel, s
 		/// <param name="label">Label.</param>
 		/// <param name="propertyExpression">Property expression.</param>
 		/// <param name="converter">Converter.</param>
-		public static LabelBinding Bind (this Label label, Expression<Func<IViewModel, object>> propertyExpression, TypeConverter converter)
+		public static LabelBinding<T> Bind<T> (this Label label, Expression<Func<IViewModel, T>> propertyExpression, TypeConverter converter)
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs]: T should have meaningful name

File: VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs [hunk 8]
```diff
@@ -148,10 +148,10 @@ public static CheckBoxBinding Bind (this CheckButton checkButton, Expression<Fun
 		/// Bind the specified SpinButton to a property by name.
 		/// </summary>
 		/// <param name="SpinButton">SpinButton.</param>
-		public static SpinBinding Bind (this SpinButton spinButton, Expression<Func<IViewModel, object>> propertyExpression,
+		public static SpinBinding<T> Bind<T> (this SpinButton spinButton, Expression<Func<IViewModel, T>> propertyExpression,
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs]: T should have meaningful name

File: VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs [hunk 9]
```diff
@@ -148,10 +148,10 @@ public static CheckBoxBinding Bind (this CheckButton checkButton, Expression<Fun
...
 										TypeConverter converter = null)
 		{
-			return new SpinBinding (spinButton, propertyExpression, converter);
+			return new SpinBinding<T> (spinButton, propertyExpression, converter);
```
Comment (ylatuya) [inline VAS.UI.Helpers.Gtk2/Bindings/BindingsExtensions.cs]: T should have meaningful name

File: VAS.UI.Gtk2/UI/Bindings/BindingsExtensions.cs [hunk 10]
```diff
@@ -18,23 +18,16 @@
...
 using VAS.Core.Interfaces.MVVMC;
-using VAS.Core.MVVMC;
 using VAS.UI.UI.Component;
 
 namespace VAS.UI.UI.Bindings
 {
 	public static class BindingsExtensions
 	{
-		public static DatePickerBinding Bind (this DatePicker widget, Expression<Func<IViewModel, DateTime>> propertyExpression)
+		public static DatePickerBinding<T> Bind<T> (this DatePicker widget, Expression<Func<IViewModel, T>> propertyExpression, TypeConverter converter = null)
```
Comment (ylatuya) [inline VAS.UI.Gtk2/UI/Bindings/BindingsExtensions.cs]: T should have meaningful name

File: VAS.Tests/MVVMC/TestOneWayPropertyBinding.cs (hunk 2 above)
Comment (VoldarGames) [reply VAS.Tests/MVVMC/TestOneWayPropertyBinding.cs]: As we talked i will add a new implementation for our Int32Converter

**Pull request merged**